lib/libmbedtls/core/cmac.c

```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2018, ARM Limited
+ * Copyright (C) 2019, Linaro Limited
+ * Copyright (C) 2021, SumUp Services GmbH
+ */
+
+#include <assert.h>
+#include <crypto/crypto.h>
+#include <crypto/crypto_impl.h>
+#include <kernel/panic.h>
+#include <mbedtls/cipher.h>
+#include <mbedtls/cmac.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tee_api_types.h>
+#include <utee_defines.h>
+#include <util.h>
+
+struct mbed_cmac_ctx {
+	struct crypto_mac_ctx mac_ctx;
+	mbedtls_cipher_context_t cipher_ctx;
+};
+
+static const struct crypto_mac_ops mbed_des3_cmac_ops;
+static const struct crypto_mac_ops mbed_aes_cmac_ops;
+
+static struct mbed_cmac_ctx *to_cmac_ctx(struct crypto_mac_ctx *ctx, const mbedtls_cipher_id_t cipher_id)
```

Comment:
Coding style: always try to limit line length to 80 characters, write as many parameters as possible on each line, and prefer alignment on the opening parenthesis (although it is also acceptable to indent less if line length would be exceeded).
```C
static struct mbed_cmac_ctx *to_cmac_ctx(struct crypto_mac_ctx *ctx,
					 const mbedtls_cipher_id_t cipher_id)
```